packages/babel-helper-member-expression-to-functions/src/index.js

```diff
@@ -100,6 +100,20 @@ const handle = {
       return;
     }
 
+    // { KEY: MEMBER } = OBJ -> { KEY: _destructureSet(MEMBER) } = OBJ
+    // [MEMBER] = ARR -> [_destructureSet(MEMBER)] = ARR
+    // [...MEMBER] -> [..._destructureSet(MEMBER)]
+    // {...MEMBER} -> {..._destructureSet(MEMBER)}
+    if (
+      (parentPath.isObjectProperty({ value: node }) &&
+        parentPath.parentPath.isObjectPattern()) ||
+      parentPath.isArrayPattern() ||
+      parentPath.isRestElement()
```

Comment:
Do we also need to check for `AssignmentPattern`s?

```js
[this.#foo = 2] = []
```

Reply:
checked and added more test case